src/validator.js

```diff
@@ -409,7 +398,7 @@ export default class Validator {
         } else {
           this.handleError(err);
         }
-        return Promise.reject(err);
+        throw err;
```

Comment:
I will say that I _do_ like throwing the errors rather than rejecting the Promises. Nobody uses rejections directly anyway ^_^